src/core.js

```diff
@@ -210,7 +210,8 @@ const matches = function(nodeName, key, matchNode) {
  * @param {?string=} key The key used to identify this element.
  */
 const alignWithDOM = function(nodeName, key) {
-  if (currentNode && matches(nodeName, key, currentNode)) {
+  if (currentNode &&
+      matches(nodeName, /** @type {Element} */ (currentNode), key)) {
```

Comment:
Oops! matches should take a `Node` not an `Element`, must have let that slip through in a previous change. Can you change it to `Node` instead?